test/__tests__/user_transformation_errors.test.js

```diff
@@ -0,0 +1,253 @@
+const { when } = require("jest-when");
+
+jest.mock("node-fetch");
+const fetch = require("node-fetch", () => jest.fn());
+
+const {
+  userTransformHandler
+} = require("../../src/util/customTransformer");
+
+const events = [
+  {
+    message: {
+      event: "event-1"
+    }
+  }
+]
+
+const workspaceId = "workspaceId";
+const versionId = "versionId";
+
+const contructTrRevCode = (code) => {
+  return {
+    codeVersion: "1",
+    language: "javascript",
+    testName: "jsTest",
+    code,
+    workspaceId,
+    versionId,
+  };
+};
+
+describe("JS Transformation Error Tests", () => {
+  describe("Transformations with transformEvent function", () => {
+    it("semantic error in base transformation - shows up under transformedEvents", async () => {
+      const code = "export function transformEvent(event, meta) {\n return events; \n}";
+      const trRevCode = contructTrRevCode(code);
+
+      const result = await userTransformHandler(
+        events,
+        versionId,
+        [],
+        trRevCode,
+        true,
+      );
+            
+      expect(result.transformedEvents[0].error).toEqual(
+        "ReferenceError: events is not defined\n    at transformEvent (base transformation:2:2)"
+      )
```

Comment:
```suggestion
      expect(result.transformedEvents.length).toBe(3);
      result.transformedEvents.forEach(ev => {expect(ev.error).toEqual(
        "ReferenceError: events is not defined\n    at transformEvent (base transformation:2:2)"
      )})
```
nit: prefer this way. validating output events count as well as each event